divvy/compute.py

```diff
@@ -220,15 +223,20 @@ def update_packages(self, config_file):
         """
         with open(config_file, 'r') as f:
             _LOGGER.info("Loading divvy config file: %s", config_file)
-            env_settings = yaml.load(f)
+            env_settings = yaml.load(f, SafeLoader)
             _LOGGER.debug("Parsed environment settings: %s",
                           str(env_settings))
 
+            old_compute_key = "compute"
+            new_compute_key = "compute_packages"
+
             # Any compute.submission_template variables should be made
             # absolute, relative to current divvy configuration file.
-            if "compute" in env_settings:
-                _LOGGER.warning("In your divvy config file, please use 'compute_packages' instead of 'compute'")
-                env_settings["compute_packages"] = env_settings["compute"]
+            if old_compute_key in env_settings:
+                warnings.warn("Divvy compute configuration '{}' section changed "
+                              "to '{}'".format(old_compute_key, new_compute_key),
+                              DeprecationWarning)
+                env_settings[new_compute_key] = env_settings[old_compute_key]
```

Comment:
This way of handling the deprecation/backcompat allows the keys to map to different values.